ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs

```diff
@@ -64,6 +64,38 @@ as |key value|}}
         </search.Select>
       </search.Search>
     </:search>
+    <:filter as |search|>
+      <search.Select
+        class="type-state"
+        @position="left"
+        @onchange={{action @filter.state.change}}
+        @multiple={{true}}
+      as |components|>
+        <BlockSlot @name="selected">
+          <span>
+            {{t "components.consul.peer.search-bar.state.name"}}
+          </span>
+        </BlockSlot>
+        <BlockSlot @name="options">
+  {{#let components.Optgroup components.Option as |Optgroup Option|}}
+    {{#each (array
+      'pending'
+      'establishing'
+      'active'
+      'failing'
+      'terminated'
+      'deleting'
```

Comment:
Def not blocking but I wonder if an enum could help here as a single source of truth for these states (to ensure they are in sync between here and predicates/peer for example)?

Reply:
Yeah agree completely, I tried to mostly centralise these in the model (I added an exported `schema`) in there, but as I'm mostly implementing (i.e. copy/pasting how its done elsewhere) I didn't put to much thought into doing anything extra. Agree its not a blocker for this, but let me have a think about this also 🤔

Reply:
I reused our `schema.State.allowedValues` array here (imported from our model files), just waiting for tests to pass again, once thats done I'll merge away